index.js

```diff
@@ -0,0 +1,33 @@
+var validationApp = angular.module("ACNApp", []);
+
+validationApp.controller("mainController", function ($scope) {
+  $scope.onSubmit = function (isValid) {
+    if (isValid) {
+      alert("ACN is valid");
+    }
+  };
+
+  $scope.onChangeAcn = function (acn) {
+    const pattern = /^(\d{3} \d{3} \d{3}|\d{9})$/i;
+    const isFormatValid = acn.match(pattern) ? true : false;
+    $scope.form.acn.$setValidity("format", isFormatValid);
+    if (!isFormatValid) return false;
+
+    acn = acn.replaceAll(" ", "");
+    let totalWeight = 0;
+    let weight = 8;
+    for (let i = 0; i < acn.length - 1; i++) {
+      const digit = acn[i];
+      totalWeight += digit * weight;
+      weight--;
+    }
+
+    const remainder = totalWeight % 10;
+    let complement = 10 - remainder;
+    if (complement == 10) complement = 0;
+
+    const isACNValid = acn[8] == complement;
+    $scope.form.acn.$setValidity("acnValid", isACNValid);
+    return isACNValid;
+  };
```

Comment:
Hi Gung De, this is looking really good.

a feedback on this function, it is not about the logic or output but this is about separation of concern, in MVC, controller accepts input and converts it to commands for the model or view and model directly manages the data, logic and rules of the application. In angularjs we consider service is a model. This function is doing the business logic as well, we need to move it into a service function. The function name also doesn't express what it is doing.

Reply:
ok thanks
should i change `index.js` file to `index.controller.js` too?

Reply:
@ellyxc I fixed it
please review it again

Note: the `onChangeAcn` function is listening to the ACN input box changes (on key up). And now everytime this function trigger, it will call `isAcnValid` function from service